prime-router/src/main/kotlin/tokens/AuthenticationStrategy.kt

```diff
@@ -1,24 +1,101 @@
+         * Even if local, if the [accessToken] is there, then do real Okta auth.
+         * @return true if we should do 'local' auth, false if we should do Okta auth.
+         */
+        fun isLocal(accessToken: String?): Boolean {
+            return when {
+                (!Environment.isLocal()) -> false
+                (accessToken != null && accessToken.split(".").size == 3) -> {
+                    // For testing auth.  Running local, but test using the real production parser.
+                    // The above test is purposefully simple so that we can test all kinds of error conditions
+                    // further downstream.
+                    logger.info("Running locally, but will use the OktaAuthenticationVerifier")
+                    false
+                }
+                else -> true
             }
+        }
 
-            // default is TokenAuthentication
-            return TokenAuthentication(DatabaseJtiCache(db))
+        /**
+         * Utility function to extract and @return the bearer access token from the [request] Authorization header,
+         * if there is one. Otherwise return null.  Fully case insensitive.
+         */
+        fun getAccessToken(request: HttpRequestMessage<String?>): String? {
+            // RFC6750 defines the access token
+            val caseInsensitiveHeaders = request.headers.mapKeys { it.key.lowercase() }
+            val authorization = caseInsensitiveHeaders[HttpHeaders.AUTHORIZATION.lowercase()] ?: return null
+            val tok = authorization.replace("Bearer ", "", ignoreCase = true)
+            return tok.ifBlank { null }
+        }
+
+        /**
+         * Authenticate a caller (which could be a human or machine).
+         * This does not perform authorization!
+         *
+         * This is a wrapper around two mechanisms for authentication.  Depending on the [request], this will do either
+         * 1) OktaAuthentication to confirm the token is a valid okta token, or,
+         * 2) two-legged (aka 'Token') authentication to confirm the token is a valid two-legged token.
+         *
+         * Two-legged uses the [db]
+         *
+         * Since the two mechanisms were developed separately, they take those slighty different arguments, so there
+         * remains some cleanup work:
+         * todo:  Have both okta and token auth create the same claims obj.  Currently we hack the token auth claims
+         * to look like okta claims, below.
+         *
+         * @return a valid claims obj if authenticated.   Otherwise null if authentication failed.
+         */
+        fun authenticate(request: HttpRequestMessage<String?>): AuthenticatedClaims? {
+            return when (request.headers["authentication-type"]) {
+                DO_OKTA_AUTH -> { // Humans using Okta will send "authentication-type": "okta" in the request header
+                    val oktaClaims = OktaAuthentication.authenticate(request) ?: return null
+                    logger.info(
```

Comment:
Change to debug or trace

Reply:
Per other comments, keeping this as is, to ensure log info is available in case of a security issue.